lib/msf/core/exploit/remote/smb/relay/target_list.rb

```diff
@@ -6,15 +6,16 @@ class TargetList
     include MonitorMixin
 
     # @param [String] targets
-    def initialize(protocol, port, targets, path=nil, randomize_targets: true)
+    def initialize(protocol, port, targets, path=nil, randomize_targets: true, drop_mic_and_flags: false)
       super()
 
       targets = Rex::Socket::RangeWalker.new(targets).to_enum(:each_ip).map do |target_ip|
         Target.new(
           ip: target_ip,
           port: port,
           protocol: protocol,
-          path: path
+          path: path,
+          drop_mic_and_flags: drop_mic_and_flags
```

Comment:
Would it be possible to split `mic` and `flags` here.

Reply:
Due to some limitation of the `Net::NTLM` library, I ended up using this instead:
```
      drop_mic_only: true,
      drop_mic_and_sign_key_exch_flags: true
```
See my comment [here](https://github.com/rapid7/metasploit-framework/pull/19832#discussion_r1993616767)